### DISCUSSION TITLE
Entity senza chiave primaria?

### DISCUSSION THREAD

**Discussion by ventoh** [fattureincloud/api#64] [Q&A] [answered]
Salve, ho una domanda: come è possibile che ci siano alcune Entity (associate a fatture) senza l'id ? Qual è allora la chiave primaria su cui fare affidamento?
Grazie

**Comment by emajo** [comment, marked as answer]
Ciao @ventoh, nel momento in cui si crea un documento si hanno due modi per settare l'entity, la prima è quella di specificare l'id di un entity già presente in anagrafica mentre la seconda è quella di specificare solamente i dati del cliente/fornitore.
Se un documento viene creato nella seconda modalità l'entity non verrà salvata in anagrafica (quindi non recuperabile dalla chiamata List Clients/Suppliers) per cui non avrà un id effettivo ma rimarrà legata solamente al documento creato.

Resto a disposizione per ulteriori chiarimenti,
Emanuele
